### ISSUE TITLE
`python.use_venv = True` does not cause packages to be installed in venv

### ISSUE THREAD

**Issue by AnOctopus** [pdm-project/pdm#973]
- [x] I have searched the issue tracker and believe that this is not a duplicate.

**Make sure you run commands with `-v` flag before pasting the output.**

## Steps to reproduce

1. `pyenv exec python -m venv .venv; source .venv/bin/activate`
2. `pdm config python.use_venv true`
3. `pdm install`

## Actual behavior

After 2, `pdm config` indicates that key is set to true
After 3, outputs `Synchronizing working set with lock file: 151 to add, 0 to update, 0 to remove`, then a bunch of successful package installs
`ls .venv/lib/python3.9/site-packages` outputs only folders for pip and setuptools
`ls __pypackages__/3.9/lib/` outputs all the folders and symlinks that should have been in the venv location.

## Expected behavior

Packages are installed into the active venv

## Environment Information

```bash
# Paste the output of `pdm info && pdm info --env` below:
PDM version:        1.13.3
Python Interpreter: /home/amanda/.pyenv/versions/3.9.9/bin/python3 (3.9)
Project Root:       /home/amanda/streamlit/streamlit
Project Packages:   /home/amanda/streamlit/streamlit/__pypackages__/3.9
{
  "implementation_name": "cpython",
  "implementation_version": "3.9.9",
  "os_name": "posix",
  "platform_machine": "x86_64",
  "platform_release": "5.16.10-arch1-1",
  "platform_system": "Linux",
  "platform_version": "#1 SMP PREEMPT Wed, 16 Feb 2022 19:35:18 +0000",
  "python_full_version": "3.9.9",
  "platform_python_implementation": "CPython",
  "python_version": "3.9",
  "sys_platform": "linux"
}
```

**Comment by frostming** [comment]
Clear the saved interpreter in `.pdm.toml` and try again.

**Comment by AnOctopus** [comment]
Seems to have worked

**Comment by JohnGurin** [comment]
Should be in documentation. Or better `pdm install` should ask what to do in such case.

**Comment by mforbes** [comment]
I second the need for better documentation. When I ran `pdm config` and saw `python.use_venv = True` I expected that PDM would use a venv as the command says, and as stated in [the docs](https://pdm.fming.dev/latest/usage/configuration/#available-configurations):

> `python.use_venv`: Install packages into the activated venv site packages instead of PEP 582

It is extremely confusing when this configuration is listed as true (or forced to be true with `PDM_USE_VENV=true`) yet a hidden `.pdm.toml` file overrides this behaviour.  Several things should be done:

1. Update the documentation to mention the role played by `.pdm.toml`.
2. Have `pdm config` check `.pdm.toml`, and if there is an interpreter there that will prevent using a venv, then set this flag to `false` in the output with an explanation that it is overridden by the interpreted in `.pdm.toml`.

Please let me know if you would like me to open a new issue.

**Comment by mforbes** [comment]
@frostming I read that.  It tells me that if the interpreter stored in `.pdm.toml` is from a virtualenv, then virtual environments will be used.  It does **not** say that, if the interpreter stored in `.pdm.toml` points elsewhere that a virtual environment will **not** be used.

The documentation under configurations and the information printed by `pdm config` explicitly says that `python.use_venv=true` will use **venv site packages instead of PEP 582**, but this is not the case. 

For me, this latter explicit promise in configurations overrode the potential implicit statement you refer to and I made me discard it as relevant.

I spent over 2 hours in earnest trying to figure out this issue from the documentation and CLI.  For people like me, the current documentation is very confusing.  (Granted, I was seriously trying to figure out PDM from the docs rather then searching for issues etc. as a test case... from the issues it is clear what to do.  But I don't think new users should be expected to delve into the issues to figure out how to use the tool.)

Happy to try to submit a doc PR, but I really am just getting started understanding `pdm`, so am not in a very good position yet to explain what should happen.
